CRM/MembershipExtras/Hook/Post/MembershipPayment.php

```diff
@@ -51,10 +51,15 @@ public function __construct($operation, $objectId, CRM_Member_DAO_MembershipPaym
   }
 
   private function setRelatedRecurContributionId() {
-    $this->relatedRecurContributionId = civicrm_api3('Contribution', 'getvalue', [
-      'id' => $this->membershipPayment->contribution_id,
-      'return' => 'contribution_recur_id',
-    ]);
+    try {
+      $this->relatedRecurContributionId = civicrm_api3('Contribution', 'getvalue', [
```

Comment:
@MiyaNoctem  as discussed, perhaps we should call API and check count instead of using try {} catch here